apps/financial-aid/backend/src/app/modules/application/application.service.ts

```diff
@@ -218,6 +218,45 @@ export class ApplicationService {
     return application
   }
 
+  async findByApplicationSystemId(
+    id: string,
+  ): Promise<ApplicationModel | null> {
+    const application = await this.applicationModel.findOne({
+      where: { applicationSystemId: id },
+      include: [
+        {
+          model: ApplicationEventModel,
+          as: 'applicationEvents',
+          separate: true,
+          order: [['created', 'DESC']],
+        },
+        {
+          model: ApplicationFileModel,
+          as: 'files',
+          separate: true,
+          order: [['created', 'DESC']],
+        },
+        {
+          model: AmountModel,
+          as: 'amount',
+          include: [{ model: DeductionFactorsModel, as: 'deductionFactors' }],
+          separate: true,
+          order: [['created', 'DESC']],
+        },
+        {
+          model: DirectTaxPaymentModel,
+          as: 'directTaxPayments',
+        },
+      ],
+    })
+
+    if (application?.amount) {
+      application.setDataValue('amount', application.amount['0'])
```

Comment:
hmm this is interesting, do we need this? 🤔

Reply:
Yes this is necessary. When we use "include" in the query then the included results are always an array and here we need to turn that array into an object because Amount in Application is not an array.